radar/2024-12-01/pulumi.md

```diff
@@ -0,0 +1,28 @@
+---
+title: "Pulumi"
+ring: hold
+quadrant: tools
+featured: true
+---
+
+Pulumi is an open-source Infrastructure as Code (IaC) tool that enables developers to define and manage cloud infrastructure using familiar programming languages like TypeScript, Python, Go, and C#. It bridges the gap between development and operations by allowing code-based infrastructure definitions, promoting reusability and collaboration.
+
+### Previous Benefits of Pulumi
+
+- **Programming Language Support:** Allowed the use of general-purpose programming languages for infrastructure definitions, providing rich abstraction capabilities.
+
+- **Multi-Cloud Compatibility:** Supported all major cloud providers, facilitating a unified approach to managing diverse environments.
+
+- **Enhanced Flexibility:** Offered advanced features like conditional logic and loops, enabling more dynamic infrastructure configurations.
+
+### Reasons for Holding
+
+- **Complexity for Teams:** The use of full-fledged programming languages introduced a steeper learning curve for infrastructure teams accustomed to declarative languages.
+
+- **Ecosystem Maturity:** Compared to Terraform, Pulumi has a smaller community and fewer pre-built modules, affecting the speed of development and issue resolution.
+
+- **Standardization Needs:** To streamline our IaC practices, standardizing on a single tool with broad industry support became a priority.
+
+### Current Stance at INFO
+
+While Pulumi brings innovative approaches to IaC, we are placing it on hold as we shift our focus to Terraform, which better aligns with our standardization and scalability goals. Existing Pulumi projects will be maintained, but future infrastructure developments will prioritize Terraform to leverage its market leadership and extensive ecosystem.
```

Comment:
Ok, agreed!

PS: we are of course also using 'plain' GitHub Workflows for IaC without any Terraform or otherwise. Good to have a discussion on this with our DevOpers.